core/constants.py

```diff
@@ -94,17 +96,21 @@ def get_package_file_contents(package: str, filepath: str) -> str:
         return file_data.decode('utf-8')
 
 
+# Here we use MyPy ignore because here error thrown by MyPy is
+# 'Missing type parameters for type "dict"' but here we don't need
+# to specify type parameters because we are inheriting dict only
+# for implementation purpose.
```

Comment:
There is a flag (--disallow-any-generics) in MyPy which takes care of this thing:

Explanation about flag -- If this flag is false then any generic type ( types that can accept type parameters like List 
 because in List[str] str is a type parameter ) is forced to have type parameters. And if this flag is True then we can use generic types without type parameters and in this case, type parameters are considered as Any:

```python
def foo(x: list) -> list    # This is allowed if --disallow-any-generics is True and here list is equivalent to List[Any]
   return ['x', 'y']

def foo(x: list) -> list:    # This is not allowed and throw an error if --disallow-any-generics is False and Here MyPy will ask for type parameters like -- list[str], list[int]
   return ['x', 'y']
```

By default -disallow-any-generics is false, so alone dict is not allowed and MyPy asks for type argument like dict[str, str]. 

Can we set the true to -disallow-any-generics flag?
We can but I am completely against it, because we if allow -disallow-any-generics then dict is equivalent to Dict[Any, Any]. So, in the future, someone can easily use dict and pass the error.

Reply:
comment changed!